### DISCUSSION TITLE
"Convert" Cycles Materials

### DISCUSSION THREAD

**Discussion by shankarsivarajan** [bnpr/Malt#374] [Feedback & Feature Requests]
For simple Cycles materials, say, Principled BSDFs with single colors or image textures, it would be nice if there were a way to "convert" those into Malt materials by, for example, using the same colors or textures and adding an outline.

This is inspired by LuxCore's "Use Cycles Settings" button. I realize the procedure is less underdetermined for physically correct rendering, but something like this would still be useful.

Here, I illustrate this with the BMW and Classroom example scenes, with materials "converted" manually.
![bmw_malt](https://user-images.githubusercontent.com/16606427/183461706-53bd5f45-bc27-4ffa-8f80-4c963b11e466.png)
![classroom_malt](https://user-images.githubusercontent.com/16606427/183461721-2851f1fa-1786-4ea3-8e47-3508688ce6fd.png)

**Comment by Kolupsy** [comment]
Interesting concept. It can definitely be done but Malt does not have an equivalent node to the Principled BSDF yet. Maybe once that is the case it would make more sense.

**Reply by shankarsivarajan** [reply]
What do you mean? If Malt is intended for NPR, whether or not it has a Principled BSDF node isn't relevant; just getting the colors from an existing Cycles scene should be enough.
I suppose you'd also want to similarly "convert" the lighting, but that should be just as easy.

**Comment by HSDSZ** [comment]
There are already a lot of high quality scene made with blender default node tree. So it will be extremely helpful if we could convert some simple blender node to malt node tree. through a script or addon?

Most resources does not have very complex blender node tree. They just have a principle bsdf and several images directly connected to them. Maybe we could get the images one by one and connect them to the equivalent sockets in MALT node tree.

By the way, do you mind sharing your manually converted classroom example file? I hope to take a deeper look

**Reply by shankarsivarajan** [reply]
Sure, but don't expect too much: [classroom_malt.zip](https://github.com/bnpr/Malt/files/9331591/classroom_malt.zip). You'll need the textures from the original classroom scene: [classroom.zip](https://download.blender.org/demo/test/classroom.zip)
